src/app/services/display.service.ts

```diff
@@ -25,7 +30,67 @@ export class DisplayService implements OnDestroy {
         return this._diagram$.getValue();
     }
 
-    public display(net: Run): void {
+    private display(net: Run): void {
         this._diagram$.next(net);
     }
+
+    public get currentRun(): Run {
```

Comment:
Wofür brauchst du das nochmal als Wert?
Der reelle Wert liegt ja im Observable `this._diagram$`, den kannst du ja an den Stellen wo du es brauchst einfach abfragen. Es sollte nicht mehrere Orte geben wo im Grunde das gleiche gespeichert wird.
Man muss dabei nur drauf achten, dass man nur das erste emittierte Element nutzt was im Observable liegt. Beispiel für `removeCurrentRun`:

```ts
    public removeCurrentRun(): void {
        this._diagram$.pipe(first())
            .subscribe((run) => this.removeRun(run));
    }
```

(In dem Rahmen kannst du `_diagram$` auch gerne zu `_currentRun$` umbenennen)

Reply:
Stimmt, das ist dann wirklich überflüssig. 
Ich habe `_diagram$ `zu `_currentRun$` umbenannt.

Mit BehaviorSubjects habe ich noch nicht so viel gearbeitet, aber könnte man statt
```
public removeCurrentRun(): void {
        this._diagram$.pipe(first())
            .subscribe((run) => this.removeRun(run));
    }
```

auch einfach
```
public removeCurrentRun(): void {
        this.removeRun(this.currentRun);
    }

public get currentRun(): Run {
        return this._currentRun$.getValue();
    }
```
verwenden oder ist da ein Unterschied? Ich möchte immer den aktuell angezeigt/ausgewählte Run löschen.

Reply:
Ah tatsächlich, wieder was gelernt 😄 
Kannte die Methode `getValue()` beim `BehaviorSubject` gar nicht, weil ich das sonst immer in Attribute vom Typ `Subject` kapsel.
Das macht den Code auf jeden Fall kleiner und man hat trotzdem einen "Single point of truth". Gefällt mir 👍🏻